realm/realm-library/src/main/java/io/realm/internal/TableView.java

```diff
@@ -370,7 +370,7 @@ public void setDouble(long columnIndex, long rowIndex, double value){
     @Override
     public void setDate(long columnIndex, long rowIndex, Date value){
         if (parent.isImmutable()) throwImmutable();
-        nativeSetDateTimeValue(nativePtr, columnIndex, rowIndex, value.getTime()/1000);
+        nativeSetTimestampValue(nativePtr, columnIndex, rowIndex, value.getTime()/1000);
```

Comment:
The `TableView` setters are not exposed in our public API, and I'm not sure if they can be called easily.